src/flamegraph/merge.rs

```diff
@@ -184,10 +184,17 @@ fn parse_nsamples(line: &mut &str) -> Option<usize> {
     let samplesi = line.rfind(' ')?;
     let mut samples = &line[(samplesi + 1)..];
 
-    // strip fractional part (if any);
+    // Strip fractional part (if any);
     // foobar 1.klwdjlakdj
-    // TODO: Properly handle fractional samples (see issue #43)
+    //
+    // The Perl version keeps the fractional part but this can be problematic
+    // because of cumulative floating point errors. Instead we recommend to
+    // use the --factor option. See https://github.com/brendangregg/FlameGraph/pull/18
     if let Some(doti) = samples.find('.') {
+        // Warn if we're stripping a non-zero fractional part.
+        if !samples[doti + 1..].chars().all(|c| c == '0') {
+            warn!("Fractional part of {} stripped. If you need to retain the extra precision you can scale up the sample data and use the --factor option to scale it back down.", samples);
```

Comment:
I wonder if we want something in place to ensure that we only print this warning _once_.

Reply:
Yes, good point. If there's one fractional sample, there's probably many. No need to repeat the same warning message over and over.

This brings up another point. We currently don't include a `log` implementation in our binaries so nothing happens with these messages. Do you want me to add one? Do you have a preferred one?

Reply:
I don't have a preferred one, but you're right that adding a `log` impl to the binary is probably a good idea (same with `collapse`).

Reply:
OK, I'll add a `log` impl in a separate PR.